src/sage/env.py

```diff
@@ -550,6 +550,4 @@ def sage_data_paths(name: str | None) -> set[str]:
     else:
         paths = {path for path in SAGE_DATA_PATH.split(os.pathsep)}
 
-    if name is None:
-        return {path for path in paths if path}
-    return {os.path.join(path, name) for path in paths if path}
+    return {os.path.join(path, name) for path in paths if os.path.exists(path)}
```

Comment:
@tobiasdiez was this the original intended behavior here? Currently `bool(path)` is always true AFAICT, so I don't see the point of this filtering.

Reply:
I think it was mostly to filter out possible `None` paths